### PR TITLE
Feature/pending major

### PR DESCRIPTION
**Pull request by ezturner**

### CODE REVIEW

File: myuw/dao/enrollment.py [hunk 1]
```diff
@@ -48,6 +71,77 @@ def get_main_campus(request):
...
+
+def _compare_degrees(first, second):
+    """
+    Takes in two lists of degrees (either major or minors) and checks to see
+    if they are the same. Returns True if so, False if they are different
+    """
+    if len(first) != len(second):
+        return False
+
+    for entry in first:
```
Comment (vegitron) [inline myuw/dao/enrollment.py]: Use a dictionary here, so you can iterate over list1, then list2, vs doing list1 * list2.

File: myuw/dao/enrollment.py [hunk 2]
```diff
@@ -48,6 +71,77 @@ def get_main_campus(request):
...
+                len(getattr(enrollments[term], accessor)) > 0):
+            enrollment = enrollments[term]
+            entry = {}
+            entry['quarter'] = term.quarter
+            entry['year'] = term.year
+
+            term_degrees = getattr(enrollments[term], accessor)
+            entry[accessor] = []
+
+            if previous is not None:
```
Comment (vegitron) [inline myuw/dao/enrollment.py]: can you change the initial state of previous so you don't have a special case here?

File: myuw/dao/term.py [hunk 3]
```diff
@@ -126,6 +126,24 @@ def get_next_quarter(request):
     return term
 
 
+def get_current_and_next_quarters(request, num):
+    """
+    Returns the current and next num uw_sws.models.Term objects in a list
+    for the current quarter refered in the user session. Returns the next
+    num -1 quarters along with the current one.
+    """
+    if hasattr(request, 'myuw_next_quarter'):
```
Comment (vegitron) [inline myuw/dao/term.py]: this will cause you some problems if someone's setting myuw_next_quarter...

File: myuw/static/js/banner/profile.js [hunk 4]
```diff
@@ -2,6 +2,33 @@ var Profile = {
     render_upon_data: function() {
         var source = $("#profile-content").html();
         var template = Handlebars.compile(source);
+
+        // Process data to determine if there are any pending majors or minors
+        var profile_data = WSData.profile_data();
+
+        if(profile_data.term_majors && profile_data.term_majors.length > 0){
```
Comment (vegitron) [inline myuw/static/js/banner/profile.js]: can you track these 2 things in the python side, and just set them at the top level of your data structure?  i think you can use 2-4 lines or so of python to drop 26 lines here

File: myuw/static/js/card/reg_status.js [hunk 5]
```diff
@@ -165,6 +166,30 @@ var RegStatusCard = {
             return;
         }
 
+        // Retrieve pending majors and minors for this quarter, if they exist
+        var profile = WSData.profile_data();
+
+        var pending_minors = [];
+        var pending_majors = [];
+
+        for(i = 0; i < profile.term_majors.length; i++){
```
Comment (vegitron) [inline myuw/static/js/card/reg_status.js]: can you turn these for loops into a method?  pending_majors = has_pending(profile.term_majors) maybe?

File: myuw/dao/enrollment.py [hunk 6]
```diff
@@ -28,10 +31,30 @@
...
     return get_enrollment_by_regid_and_term(regid,
                                             get_current_quarter(request))
 
 
+def get_all_enrollments():
+    regid = get_regid_of_current_user()
+    return _get_all_enrollments(regid)
+
+
+def _get_all_enrollments(regid):
```
Comment (fanglinfang) [inline myuw/dao/enrollment.py]: Ethan, 
you may remove line 47-55 and use "enrollment_search_by_regid(regid)" on line 44.

File: myuw/dao/term.py (hunk 3 above)
Comment (fanglinfang) [inline myuw/dao/term.py]: Ethan, 
consider use a different key, such as "curr_and_next_quarters"

File: myuw/resources/sws/file/student/v5/enrollment.json_reg_id_10000000000000000000000000000010_verbose_true_transcriptable_course_all_changed_since_date_ [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+                    "WritingCourse": false
+                }
+            ],
+            "RepositoryTimeStamp": "3/28/2013 2:22:05 PM",
+            "Term": {
+                "Href": "/student/v5/term/2013,winter.json",
+                "Quarter": "winter",
+                "Year": 2013
+            }
+        }
```
Comment (fanglinfang) [inline myuw/resources/sws/file/student/v5/enrollment.json_reg_id_10000000000000000000000000000010_verbose_true_transcriptable_course_all_changed_since_date_]: line 138 missed a comma, which is fixed on feature/purple branch. 
https://github.com/uw-it-aca/myuw/pull/595

File: myuw/dao/enrollment.py [hunk 8]
```diff
@@ -18,18 +21,62 @@
...
+    """
+    result_dict = get_all_enrollments()
+    terms = result_dict.keys()
+    for key in terms:
+        if key not in term_list:
+            del result_dict[key]
+    return result_dict
+
+
+def get_all_enrollments():
```
Comment (fanglinfang) [inline myuw/dao/enrollment.py]: line 66 - 79 are redundant with that defined on line 28, please remove them.

File: myuw/dao/enrollment.py [hunk 9]
```diff
@@ -18,18 +21,62 @@
     "SENIOR": 4,
     "GRADUATE": 5,
 }
+DEFAULT_CLASS_CODE = 6
+logger = logging.getLogger(__name__)
 
 
-DEFAULT_CLASS_CODE = 6
+def get_all_enrollments():
```
Comment (fanglinfang) [inline myuw/dao/enrollment.py]: This function was:
def get_all_enrollments():
    """
    :return: the dictionary of {Term: Enrollment}
    """
    regid = get_regid_of_current_user()
    return enrollment_search_by_regid(regid)


line 36-37 seem unnecessary.

Comment (fanglinfang) [inline myuw/dao/enrollment.py]: line 36-27 are not necessary as enrollment_search_by_regid has been tested in uw_sws

**Pull request closed without merging**